packages/origin-247-certificate/test/offchain-certificate/certificateEventRepository.e2e-spec.ts

```diff
@@ -260,6 +246,45 @@ describe('CertificateEventRepository', () => {
 
             expect(certs).toHaveLength(1);
         });
+
+        it('should not process other events for that certificate if issue failed', async () => {
+            const internalId = 1;
+            const [issue] = await prepareEvents(
+                CertificateIssuedEvent.createNew(internalId, createIssueCommand()),
+                CertificateTransferredEvent.createNew(internalId, createTransferCommand())
+            );
+            await updateAttempt(issue, { error: 'some error', tries: 4 });
+
+            const certs = await certificateEventRepository.getAllNotProcessed();
+
+            expect(certs).toHaveLength(0);
+        });
+
+        it('should not process other transfer event for that certificate if first transfer failed', async () => {
+            const internalId = 1;
+            const [firstTransfer] = await prepareEvents(
+                CertificateTransferredEvent.createNew(internalId, createTransferCommand()),
+                CertificateTransferredEvent.createNew(internalId, createTransferCommand())
+            );
+            await updateAttempt(firstTransfer, { error: 'some error', tries: 4 });
+
+            const certs = await certificateEventRepository.getAllNotProcessed();
+
+            expect(certs).toHaveLength(0);
+        });
+
+        it('should not process other transfer event for that certificate if claim failed', async () => {
```

Comment:
Maybe I'm picky, but TBH I would test happy path in the certificateEventRepository, and all other paths in `offchain-certificate` unit test. This way it will cover cases a bit better

Reply:
I'm not sure about that. Repository has that knowledge and is executing that logic. Maybe it doesn't know what exactly these cases - it's more like `filter out all events that have previously failed for some certificate`. But `offchain-certificate` doesn't have that knowledge at all - he's just processing everything he gets from repository without knowing the internals